From 53aee89eb4696a444c5d65c3fdd26fcf4cf377ce Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Tue, 19 Aug 2014 15:27:01 -0400
Subject: [tools] perf/report: Postpone setting up browser after parsing options

Message-id: <1408462094-14194-253-git-send-email-jolsa@redhat.com>
Patchwork-id: 88274
O-Subject: [PATCH RHEL7.1 BZ1131394 252/325] perf report: Postpone setting up browser after parsing options
Bugzilla: 1131394
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1131394
https://bugzilla.redhat.com/show_bug.cgi?id=1131394

upstream
========
commit 4bceffbc26fab2444742db59c6f8124c29e41369
Author: Namhyung Kim <namhyung.kim@lge.com>
Date: Fri Nov 1 16:33:12 2013 +0900

description
===========
If setup_browser() called earlier than option parsing, the actual error
message can be discarded during the terminal reset.  So move it after
setup_sorting() checks whether the sort keys are valid.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/builtin-report.c b/tools/perf/builtin-report.c
index 98d3891..4df3161 100644
--- a/tools/perf/builtin-report.c
+++ b/tools/perf/builtin-report.c
@@ -905,13 +905,6 @@ int cmd_report(int argc, const char **argv, const char *prefix __maybe_unused)
    input_name = "perf.data";
  }
 
- if (strcmp(input_name, "-") != 0)
-  setup_browser(true);
- else {
-  use_browser = 0;
-  perf_hpp__init();
- }
-
  file.path  = input_name;
  file.force = report.force;
 
@@ -957,6 +950,18 @@ repeat:
  if (setup_sorting() < 0)
   usage_with_options(report_usage, options);
 
+ if (parent_pattern != default_parent_pattern) {
+  if (sort_dimension__add("parent") < 0)
+   goto error;
+ }
+
+ if (strcmp(input_name, "-") != 0)
+  setup_browser(true);
+ else {
+  use_browser = 0;
+  perf_hpp__init();
+ }
+
  /*
   * Only in the TUI browser we are doing integrated annotation,
   * so don't allocate extra space that won't be used in the stdio
@@ -986,11 +991,6 @@ repeat:
  if (symbol__init() < 0)
   goto error;
 
- if (parent_pattern != default_parent_pattern) {
-  if (sort_dimension__add("parent") < 0)
-   goto error;
- }
-
  if (argc) {
   /*
    * Special case: if there's an argument left then assume that
-- 
1.7.1